server/index.js

```diff
@@ -29,6 +29,34 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+  const sql = `
+    select *
+    from "products"
+    where "productId" = $1
+  `;
+
+  const params = [productId];
+  if (productId > 6 || productId <= 0) {
+    return next(new ClientError('Invalid product', 400));
+  } else if (productId === '') {
+    res.status(500).json({
+      error: 'An unexpected error occurred'
+    });
+  }
+  db.query(sql, params)
+    .then(result => {
+      return res.json(result.rows[0]);
+    })
+    .catch(err => {
+      console.error(err);
+      return res.status(404).json({
+        error: 'There may be an error querying the database.'
+      });
```

Comment:
This was an internal server error (`500`). Instead, just pass the error to `next()`

```js
.catch(err => next(err))
```